sdk/core/azure-core/CMakeLists.txt

```diff
@@ -46,6 +46,7 @@ set(
     ${CURL_TRANSPORT_ADAPTER_INC}
     ${WIN_TRANSPORT_ADAPTER_INC}
     inc/azure/core/credentials/credentials.hpp
+    inc/azure/core/credentials/token_credential_options.hpp
```

Comment:
We need to include these headers to `core.hpp` and add tests to the simplified header.
Fixed in https://github.com/Azure/azure-sdk-for-cpp/pull/2070